lib/sensu-plugin/utils.rb

```diff
@@ -33,6 +41,82 @@ def read_event(file)
         exit 0
       end
 
+      ##
+      #  Helper method to convert Sensu 2.0 event into Sensu 1.4 event
+      #    This is here to help keep Sensu Plugin community handlers working
+      #    until they natively support 2.0
+      #    Takes 2.0 event json object as argument
+      #    Returns event with 1.4 mapping included
+      #
+      #    Note:
+      #      The 1.4 mapping overwrites some attributes so the resulting event cannot
+      #      be used in a 2.0 workflow. The top level boolean attribute "v2_event_mapped_into_v1"
+      #      will be set to true as a hint to indicate this is a mapped event object.
+      #
+      ##
+      def map_v2_event_into_v1(orig_event = nil)
+        orig_event ||= @event
+
+        # return orig_event if already mapped
+        return orig_event if orig_event['v2_event_mapped_into_v1']
+
+        # Deep copy of orig_event
+        event = Marshal.load(Marshal.dump(orig_event))
+
+        # Trigger mapping code iff enity exists and client does not
+        client_missing = event['client'].nil? || event['client'].empty?
+        if event.key?('entity') && client_missing
+          ##
+          # create the client hash from the entity hash
+          ##
+          event['client'] = event['entity']
+
+          ##
+          # Fill in missing client attributes
+          ##
+          event['client']['name']        ||= event['entity']['id']
+          event['client']['subscribers'] ||= event['entity']['subscriptions']
+
+          ##
+          # Fill in renamed check attributes expected in 1.4 event
+          #   subscribers, source
+          ##
+          event['check']['subscribers'] ||= event['check']['subscriptions']
+          event['check']['source'] ||= event['check']['proxy_entity_id']
+
+          ##
+          # Mimic 1.4 event action based on 2.0 event state
+          #  action used in logs and fluentd plugins handlers
+          ##
+          state = event['check']['state'] || 'unknown::2.0_event'
+          event['action'] ||= 'flapping' if state.casecmp('flapping').zero?
```

Comment:
I feel like this could be much simpler and less error prone with a case statement.

Reply:
refactored using a hash mapping, since the states to action are 1 to 1.